lib/spack/spack/cmd/create.py

```diff
@@ -600,6 +617,9 @@ def get_build_system(args, guesser):
         else:
             msg = "This package looks like it uses the {0} build system"
             tty.msg(msg.format(template))
+    else:
+        template = 'bundle'
+        tty.warn("No URL provided.  Using the dependencies package template.")
```

Comment:
`dependencies` still here despite name change.

Reply:
Great catch.  Thanks.

Reply:
> It's hard for me to see the changes here, as everything has been factored into `PackageCore`, `PackageBase`, and `BundlePackage`, but my main question is whether it's necessary to add a new class to the hierarchy. I would rather see this done by just making it so that packages don't have to have URLs. This would be useful, among other things, for testing.
> 
> Adding a `BundlePackage` class as a completely separate case (or at least as a completely separate branch of the package hierarchy) seems awkward to me -- I do not think there is anything _particularly_ special about other packages that requires a URL -- fetching should just not work for those packages.
> 
> I actually _like_ `BundlePackage` as something you could extend (it's informative in the package description), but I'm mainly concerned that it's a completely separate subclass from `PackageBase`. Seems like it could just be a subclass of `PackageBase` like the others, we just wouldn't always require `url` in `PackageBase`

I took a more "minimal" change approach in https://github.com/spack/spack/pull/11981, which is ready for review.